nuxeo-ai-core/src/main/java/org/nuxeo/ai/enrichment/EnrichingStreamProcessor.java

```diff
@@ -196,20 +202,31 @@ protected Collection<? extends AIMetadata> cacheGet(String cacheKey) {
          * Calls the provider using the retryPolicy
          */
         protected Collection<AIMetadata> callProvider(Record record, Callable<Collection<AIMetadata>> callable) {
-            return Failsafe.with(retryPolicy).onSuccess(r -> {
-                metrics.success();
-                if (log.isDebugEnabled()) {
-                    log.debug("Enrichment result is " + r);
-                }
-            }).onFailedAttempt(failure -> {
-                metrics.error();
-                log.warn("Enrichment error ({}) for record: {} ", enricherName, record, failure);
-            }).onRetry(c -> {
+
+            CheckedSupplier<Collection<AIMetadata>> supplier = callable::call;
+            Timeout<Collection<AIMetadata>> timeout = Timeout.of(Duration.ofSeconds(60));
+
+            retryPolicy.onRetry(c -> {
```

Comment:
I'm not sure this is what we want, as I understand it the given listener will be registered to the `retryPolicy` and not the ongoing failsafe execution, the retry policy should be copied.

See next comment.